CMakeLists.txt

```diff
@@ -26,7 +26,7 @@ include_directories(
 )
 
 find_package(Eigen3 REQUIRED)
-include_directories(${EIGEN3_INCLUDE_DIR})
+include_directories(SYSTEM ${EIGEN3_INCLUDE_DIR})
```

Comment:
Yes, this is really useful for isolating warnings from third party libraries (where we can't fix them) versus our code (which we should fix warnings in).